{"original_position": 30, "diff_hunk": "@@ -28,6 +31,46 @@ def test_it_works(self, rf):\n             'is_approved': recipe.is_approved\n         }\n \n+    def test_it_validates_arguments(self):\n+        schema = json.loads(\n+            open('normandy/recipes/static/actions/console-log/package.json').read()\n+        )\n+\n+        ActionFactory(\n+            name='console-log',\n+            arguments_schema=schema['normandy']['argumentsSchema']\n+        )\n+\n+        # If the action specified cannot be found, raise validation\n+        # error indicating the arguments schema could not be loaded\n+        serializer = RecipeSerializer(data={\n+            'action': 'action-that-doesnt-exist', 'arguments': {}\n+        })\n+\n+        serializer.is_valid()\n+        assert serializer.errors['arguments'] == ['Could not find arguments schema.']\n+        assert pytest.raises(serializers.ValidationError)", "body_text": "I don't think this does what you want it to. Usually this is used as\nwith pytest.raises(SomeExceptionType):\n    something_that_raises_that_type()\nJust calling pytest.raises(...) alone doesn't actually assert anything.", "pull_request_url": "https://api.github.com/repos/mozilla/normandy/pulls/196", "url": "https://api.github.com/repos/mozilla/normandy/pulls/comments/71242145", "created_at": "2016-07-18T22:24:30Z", "author_association": "MEMBER", "body": "I don't think this does what you want it to. Usually this is used as\n\n``` python\nwith pytest.raises(SomeExceptionType):\n    something_that_raises_that_type()\n```\n\nJust calling `pytest.raises(...)` alone doesn't actually assert anything.\n", "updated_at": "2016-07-19T14:15:13Z", "html_url": "https://github.com/mozilla/normandy/pull/196#discussion_r71242145", "pull_request_review_id": null, "_links": {"self": {"href": "https://api.github.com/repos/mozilla/normandy/pulls/comments/71242145"}, "html": {"href": "https://github.com/mozilla/normandy/pull/196#discussion_r71242145"}, "pull_request": {"href": "https://api.github.com/repos/mozilla/normandy/pulls/196"}}, "commit_id": "307546e51d66ab4f9a66874cf756a949629c932a", "user": {"following_url": "https://api.github.com/users/mythmon/following{/other_user}", "events_url": "https://api.github.com/users/mythmon/events{/privacy}", "organizations_url": "https://api.github.com/users/mythmon/orgs", "url": "https://api.github.com/users/mythmon", "gists_url": "https://api.github.com/users/mythmon/gists{/gist_id}", "html_url": "https://github.com/mythmon", "subscriptions_url": "https://api.github.com/users/mythmon/subscriptions", "avatar_url": "https://avatars3.githubusercontent.com/u/305049?v=4", "repos_url": "https://api.github.com/users/mythmon/repos", "received_events_url": "https://api.github.com/users/mythmon/received_events", "gravatar_id": "", "starred_url": "https://api.github.com/users/mythmon/starred{/owner}{/repo}", "site_admin": false, "login": "mythmon", "type": "User", "id": 305049, "followers_url": "https://api.github.com/users/mythmon/followers"}, "position": null, "path": "normandy/recipes/tests/test_serializers.py", "body_html": "<p>I don't think this does what you want it to. Usually this is used as</p>\n<div class=\"highlight highlight-source-python\"><pre><span class=\"pl-k\">with</span> pytest.raises(SomeExceptionType):\n    something_that_raises_that_type()</pre></div>\n<p>Just calling <code>pytest.raises(...)</code> alone doesn't actually assert anything.</p>", "original_commit_id": "68369eb43b94985f24f722399663a07608e362b3", "id": 71242145}